serverlessworkflow/modules/ROOT/pages/cloud/operator/configuring-workflow-eventing-system.adoc

```diff
@@ -0,0 +1,315 @@
+= Configuring the Workflow Eventing system
+:compat-mode!:
+// Metadata:
+:description: Workflows eventing system configuration
+:keywords: kogito, sonataflow, workflow, serverless, operator, kubernetes, knative, knative-eventing, events
+
+This document describes how to configure the eventing system for a {product_name} workflow.
+
+In general, the following events are produced in a {product_name} installation:
+
+* Workflow outgoing and incoming business events.
+
+* SonataFlow system events sent from the workflow to the Data Index and Job Service respectively.
+
+* SonataFlow system events sent from the Jobs Service to the Data Index Service.
+
+The SonataFlow Operator is designed to use the link:{knative_eventing_url}[Knative Eventing] system to resolve all the event communication between these services.
+
+[NOTE]
+====
+In a regular SonataFlow installation, the preferred method is to use the <<platform-scoped-eventing-system-configuration>>, while the <<workflow-scoped-eventing-system-configuration>> is reserved only for advanced use cases.
+====
+
+[#platform-scoped-eventing-system-configuration]
+== Platform-scoped Eventing system configuration
+
+To configure a platform-scoped eventing system, you must use the field `spec.eventing.broker.ref` in the `SonataFlowPlatform` CR to refer to a Knative Eventing Broker.
+
+This information signals the {operator_name} to automatically link every workflow deployed in that namespace, to `produce` and `consume` the events by using that Broker.
+
+Additionally, the supporting services deployed in that namespace, that don't provide a custom eventing system configuration, will be linked to that Broker.
+For more information about configuring the supporting services eventing system, xref:cloud/operator/supporting-services.adoc#configuring-supporting-services-eventing-system[see].
+
+The following `SonataFlowPlatform` CR fragment shows an example of such configuration:
+
+.Platform scoped eventing system configuration example
+
+include::common/platform-scoped-eventing-system-configuration-example.adoc[]
+
+[#workflow-scoped-eventing-system-configuration]
+== Workflow-scoped Eventing system configuration
+
+A workflow-scoped eventing system configuration provides the ability to do a fine-grained configuration of the eventing system for the events `produced` and `consumed` by a workflow.
+
+To configure a workflow-scoped eventing system you must use the fields `spec.sink.ref` and `spec.sources[]` in the `SonataFlow` CR.
+
+[#outgoing-eventing-system-configuration]
+=== Outgoing Eventing system configuration
+
+To configure a workflow-scoped eventing system for the workflow outgoing events, you must use the field `spec.sink.ref` in the `SonataFlow` CR.
+
+This information signals the SonataFlow Operator to automatically link the current workflow, to produce the events by using that Broker.
```

Comment:
```suggestion
This information signals the {operator_name} to automatically link the current workflow, to produce the events by using that Broker.
```

Reply:
Not necessarily a broker, but any valid `Sink`, correct?

Reply:
yes

Reply:
but I believe that at the point we are now, and considering that we are managing only one "sink" where to write the "system events" (i.e., the events produced for the DI and JS) and the workflow "busness events", to document more complex settings, will be a bit hard to understand/manage by the users.

Reply:
In 1.36.0, or, future releases, one thing I want to explore is to have the ability of configuring different "sinks", to separate system events from business events. So that we can have.

1. A system events "sink", where all the events for DI and JS flows.

2. Potentially define different "sinks" for the business events delivered by the WF, which could be, a sort of "global sink" to where to deliver all the workflow business events, or, in a more advanced use case, people could use different "sinks" per event type.

But we need to think a bit.